app/containers/Applications/ApplicationDetailsContainer.tsx

```diff
@@ -50,17 +49,17 @@ export const ApplicationDetailsComponent: React.FunctionComponent<Props> = (
     props.relay.refetch(refetchVariables);
   }, [oldDiffVersion, newDiffVersion, props.application.id, props.relay]);
 
-  // const handleDownloadPdf = async () => {
-  //   const response = await fetch('/print-pdf', {
-  //     method: 'POST',
-  //     headers: {
-  //       'Content-Type': 'application/json'
-  //     },
-  //     body: JSON.stringify({html: document.documentElement.outerHTML})
-  //   });
-  //   const blob = await response.blob();
-  //   FileDownload(blob, 'CIIP_Application.pdf');
-  // };
+  const handleDownloadPdf = async () => {
+    const response = await fetch('/print-pdf', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({html: document.documentElement.outerHTML})
```

Comment:
@wenzowski I would assume that this is how you allow untrusted HTML, and how you get hacked?

Reply:
Using `renderToStaticMarkup`, which would fix the security issue, might be a bit tricky as the Relay containers have to be rendered within a QueryRenderer

Reply:
yes this would allow injection of whatever html a user injects into `document` via dev tools (incl malicious scripts)

Reply:
should be possible to `renderToStaticMarkup` via nextjs server-side dynamic page rendering (though if relay is needed to fetch page contents we'd need to fix the issue with relay not running on server-side-render...should be possible to fix...used to work...)